library/sequencer/src/lib/ui/sequence.rb

```diff
@@ -0,0 +1,71 @@
+require "yast"
+Yast.import "Sequencer"
+
+# FIXME: once the API is reviewed, move this to yast-yast2
+module UI
+  # A {UI::Sequence} is an object-oriented interface for the good old
+  # {Yast::SequencerClass Yast::Sequencer}.
+  # In the simple case it runs a sequence of dialogs
+  # connected by Back and Next buttons.
+  class Sequence
+    include Yast::I18n
+
+    # A drop-in replacement for
+    # {Yast::SequencerClass#Run Yast::Sequencer.Run}
+    def self.run(aliases, sequence)
+      Yast::Sequencer.Run(aliases, sequence)
+    end
+
+    # A replacement for
+    # {Yast::SequencerClass#Run Yast::Sequencer.Run}
+    # but smarter:
+    # - auto :abort (see {#abortable})
+    # - *aliases* are assumed to be method names if unpecified
+    #   (see {#from_methods})
+    def run(aliases: nil, sequence:)
+      aliases = from_methods(sequence) if aliases.nil?
+      self.class.run(aliases, abortable(sequence))
+    end
+
+    # Add !{abort: :abort} transitions if missing
+    # (an :abort from a dialog should :abort the whole sequence)
+    def abortable(sequence_hash)
+      sequence_hash.map do |name, destination|
+        if name == "ws_start"
```

Comment:
this "hack" should be documented why it is needed. Also it smells for me like constant.